tools/dexcom_analyze/dexcom_analyze.go

```diff
@@ -0,0 +1,1543 @@
+			if record.AvailableTime != nil {
+				record.AppendIssue(Issue_Task_With_State_Failed_AvailableTime_Present)
+			}
+			if record.DeadlineTime != nil {
+				record.AppendIssue(Issue_Task_With_State_Failed_DeadlineTime_Present)
+			}
+			if record.Error == nil {
+				record.AppendIssue(Issue_Task_With_State_Failed_Error_Missing)
+			}
+			if record.ExpirationTime != nil {
+				record.AppendIssue(Issue_Task_With_State_Failed_ExpirationTime_Present)
+			}
+		case task.TaskStateCompleted:
+			record.AppendIssuef(IssueFormat_Task_State_Invalid, record.State)
+		}
+
+		if record.providerSession == nil {
+			record.AppendIssue(Issue_Task_ProviderSession_Missing)
+		}
+	}
+}
+
+func (t *Tool) analyzeIssues() {
+	issueMarshalableMap := IssueMarshalableMap{}
+	for _, record := range t.providerSessionsMap {
+		issueMarshalableMap.AppendIssueReporterAsMarshalable(record)
+	}
+	for _, record := range t.dataSourcesMap {
+		issueMarshalableMap.AppendIssueReporterAsMarshalable(record)
+	}
+	for _, record := range t.tasksMap {
+		issueMarshalableMap.AppendIssueReporterAsMarshalable(record)
+	}
+	for _, record := range t.usersMap {
+		issueMarshalableMap.AppendIssueReporterAsMarshalable(record)
+	}
+
+	t.outputIssues(issueMarshalableMap)
+}
+
+func (t *Tool) outputIssues(issueMarshalableMap IssueMarshalableMap) {
+	t.outputIssuesHeader(issueMarshalableMap)
+
+	issues := maps.Keys(issueMarshalableMap)
+	sort.StringSlice(issues).Sort()
+	for _, issue := range issues {
+		t.outputIssue(issue, issueMarshalableMap[issue], issueMarshalableMap)
+	}
+
+	t.outputIssuesFooter()
+}
+
+func (t *Tool) outputIssue(issue string, marshalables Marshalables, issueMarshalableMap IssueMarshalableMap) {
+	t.outputIssueHeader(issue, marshalables)
+
+	switch issue {
+	case Issue_DataSource_ProviderSessionID_DoesNotExist:
+		t.outputResolutionHeader("Examine each to determine why it does not exist.")
+		t.outputMongoReadOperationsHeader()
+		t.outputMongoDataSourcesAggregation(marshalables.DataSources().IDs())
+		t.outputMongoWriteOperationsHeader()
+		t.outputMongoOperationf("db.data_sources.updateMany({id: {$in: [%s]}}, {$set: {state: 'disconnected'}, $unset: {providerSessionId: true}})", mongoIDs(marshalables.DataSources().IDs()))
+		return
+	case Issue_DataSource_ProviderSessionID_Empty:
+	case Issue_DataSource_UserID_Empty:
+	case Issue_DataSource_UserID_Missing:
+	case Issue_DataSource_With_DataSetIDs_DataSetIDs_Length_Invalid:
+	case Issue_DataSource_With_DataSetIDs_EarliestDataTime_Missing, Issue_DataSource_With_DataSetIDs_LastImportTime_Missing, Issue_DataSource_With_DataSetIDs_LatestDataTime_Missing:
+		t.outputResolutionHeader("FIXED with BACK-3113. Will keep occurring until deployed. Will need to manually fix.")
+		// t.outputMongoReadOperationsHeader()
+
+		// var earliestDataTimeIDs IDs
+		// if otherMarshable, ok := issueMarshalableMap[Issue_DataSource_With_DataSetIDs_EarliestDataTime_Missing]; ok {
+		// 	earliestDataTimeIDs = otherMarshable.DataSources().IDs()
+		// }
```

Comment:
all the commented out sections?

Reply:
Those are intended to provide more details, if desired. I added a NOTE comment above each to indicate this.